app/src/main/java/org/codefordenver/encorelink/LoginActivity.java

```diff
@@ -0,0 +1,105 @@
+package org.codefordenver.encorelink;
+
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.os.Handler;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.EditText;
+
+import android.widget.Button;
+import android.widget.Toast;
+
+public class LoginActivity extends AppCompatActivity {
+    private static final String TAG = "LoginActivity";
+    private static final int REQUEST_SIGNUP = 0;
+
+    private static EditText emailText;
+    private static EditText passwordText;
+    private static Button login_button;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        Login();
+    }
+
+    public void Login() {
+        emailText = findViewById(R.id.input_email);
+        passwordText = findViewById(R.id.input_password);
+        login_button = findViewById(R.id.button_login);
+
+        login_button.setOnClickListener(
+                new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        Log.d(TAG, "Login");
+
+                        if (!validate()) {
+                            onLoginFailed();
+                            return;
+                        }
```

Comment:
Validation is in the wrong spot. Need to grab user name and password first before calling validate and onLoginFailed.